TYPO3.Neos/Classes/TYPO3/Neos/Service/NodeActionsService.php

```diff
@@ -0,0 +1,135 @@
+<?php
+namespace TYPO3\Neos\Service;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "TYPO3.Neos".            *
+ *                                                                        *
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU General Public License, either version 3 of the   *
+ * License, or (at your option) any later version.                        *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+use TYPO3\Eel\FlowQuery\FlowQuery;
+use TYPO3\Eel\Utility as EelUtility;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Object\ObjectManagerInterface;
+use TYPO3\TYPO3CR\Domain\Model\NodeInterface;
+use TYPO3\TYPO3CR\Domain\Service\NodeTypeManager;
+
+/**
+ * Functions for executing actions on node creation (e.g. creating child nodes or setting properties based on wizard
+ * data)
+ *
+ * @Flow\Scope("singleton")
+ */
+class NodeActionsService
```

Comment:
A related thought: this is a domain service, not a REST service, so it should be moved as well.